src/gp_instance.cpp

```diff
@@ -174,3 +175,63 @@ void GPInstance::EstimateSBNParameters() {
   double marginal_log_lik = GetEngine()->GetLogMarginalLikelihood();
   std::cout << std::setprecision(9) << marginal_log_lik << std::endl;
 }
+
+RootedTreeCollection GPInstance::GetRootedTreeCollection() {
+
+  Tree::TreeVector tree_vector;
+  Node::NodePtrVec trees = dag_.GenerateAllTrees();
+  EigenVectorXd bl = engine_->GetBranchLengths();
+
+  // Leaves encoding to parent subsplit encoding.
+  std::unordered_map<const Node*, Bitset> leaves2subsplit_indexer;
+  for (size_t i = 0; i < trees.size(); i++) {
+    trees.at(i)->PreOrder([this, &leaves2subsplit_indexer](const Node *node)
+    {
+      GPDAGNode *dag_node = dag_.GetDagNode(node->Id());
+      if (!leaves2subsplit_indexer.count(node)) {
+        SafeInsert(leaves2subsplit_indexer, node, dag_node->GetBitset());
+      }
+
+      for (size_t i = 0; i < node->Children().size(); i++) {
+        size_t child_id = node->Children().at(i)->Id();
+        GPDAGNode *child_node = dag_.GetDagNode(child_id);
+        size_t i0 = dag_.GetGPCSPIndex(dag_node->GetBitset(false) + child_node->GetBitset());
+        size_t i1 = dag_.GetGPCSPIndex(dag_node->GetBitset(true) + child_node->GetBitset());
+        size_t gpcsp_idx = std::min(i0, i1);
+        Assert(gpcsp_idx < SIZE_MAX, "GPCSP does not exist.");
```

Comment:
So the point of this block of code is to make sure that everything exists?

Reply:
Removed this block of code.

Reply:
Removed.